eradicate.py

```diff
@@ -28,176 +28,199 @@
+
+    DEFAULT_WHITELIST = [
+        r'pylint',
+        r'pyright',
+        r'noqa',
+        r'type:\s*ignore',
+        r'fmt:\s*(on|off)',
+        r'TODO',
+        r'FIXME',
+        r'XXX'
+    ]
+    WHITELIST_REGEX = re.compile(r'|'.join(DEFAULT_WHITELIST), flags=re.IGNORECASE)
+
+    def comment_contains_code(self, line, aggressive=True):
+        """Return True comment contains code."""
+        line = line.lstrip()
+        if not line.startswith('#'):
+            return False
+
+        line = line.lstrip(' \t\v\n#').strip()
+
+        # Ignore non-comment related hashes. For example, "# Issue #999".
+        if re.search('#[0-9]', line):
+            return False
+
+        # Ignore whitelisted comments
+        if self.WHITELIST_REGEX.search(line):
+            return False
+
+        if re.match(r'.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)', line):
+            return False
+
+        # Check that this is possibly code.
+        for symbol in list('()[]{}:=%') + ['print', 'return', 'break', 'continue',
+                                           'import']:
+            if symbol in line:
+                break
+        else:
+            return False
 
-MULTILINE_ASSIGNMENT_REGEX = re.compile(r'^\s*\w+\s*=.*[(\[{]$')
-PARTIAL_DICTIONARY_REGEX = re.compile(r'^\s*[\'"]\w+[\'"]\s*:.+[,{]\s*$')
-
-
-def comment_contains_code(line, aggressive=True):
-    """Return True comment contains code."""
-    line = line.lstrip()
-    if not line.startswith('#'):
-        return False
-
-    line = line.lstrip(' \t\v\n#').strip()
-
-    # Ignore non-comment related hashes. For example, "# Issue #999".
-    if re.search('#[0-9]', line):
-        return False
-
-    if line.startswith('pylint:'):
-        return False
+        if self.multiline_case(line, aggressive=aggressive):
+            return True
 
-    if re.match(r'.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)', line):
-        return False
+        symbol_list = [r'elif\s+.*', 'else', 'try',
+                       'finally', r'except\s+.*']
+        if aggressive:
+            symbol_list.append(r'if\s+.*')
 
-    # Check that this is possibly code.
-    for symbol in list('()[]{}:=%') + ['print', 'return', 'break', 'continue',
-                                       'import']:
-        if symbol in line:
-            break
-    else:
-        return False
+        for symbol in symbol_list:
+            if re.match(r'^\s*' + symbol + r'\s*:\s*$', line):
+                return True
 
-    if multiline_case(line, aggressive=aggressive):
-        return True
+        line = re.sub(r'^(print|return)\b\s*', '', line)
 
-    symbol_list = [r'elif\s+.*', 'else', 'try',
-                   'finally', r'except\s+.*']
-    if aggressive:
-        symbol_list.append(r'if\s+.*')
+        if re.match(self.PARTIAL_DICTIONARY_REGEX, line):
+            return True
 
-    for symbol in symbol_list:
-        if re.match(r'^\s*' + symbol + r'\s*:\s*$', line):
+        try:
```

Comment:
nitpick:

```python
try:
   compile(line, '<string>', 'exec')
except (SyntaxError, TypeError, UnicodeDecodeError):
   return False
else:
   return True
```

I prefer to have `try`s with just a single line of code.

Reply:
I like that better too.